precise/iaf_psc_exp_ps_time_reversal.h

```diff
@@ -0,0 +1,589 @@
+    //@}
+    
+    // The next two classes need to be friends to access the State_ class/member
+    friend class RecordablesMap<iaf_psc_exp_ps_time_reversal>;
+    friend class UniversalDataLogger<iaf_psc_exp_ps_time_reversal>;
+    
+    void set_spiketime(Time const &);
+    
+    /**
+     * Propagate neuron state.
+     * Propagate the neuron's state by dt.
+     * @param dt Interval over which to propagate
+     */
+    void propagate_(const double_t dt);
+    
+    /**
+     * Emit a single spike caused by DC current in absence of spike input.
+     * Emits a single spike and reset neuron given that the membrane
+     * potential was below threshold at the beginning of a mini-timestep
+     * and above afterwards.
+     *
+     * @param origin  Time stamp at beginning of slice
+     * @param lag     Time step within slice
+     * @param t0      Beginning of mini-timestep
+     * @param dt      Duration of mini-timestep
+     */
+    void emit_spike_(const Time & origin, const long lag, 
+		     const double_t t0, const double_t dt);
+    
+    /**
+     * Emit a single spike at a precisely given time.
+     *
+     * @param origin        Time stamp at beginning of slice
+     * @param lag           Time step within slice
+     * @param spike_offset  Time offset for spike
+     */
+    void emit_instant_spike_(const Time & origin, const long lag, 
+			     const double_t spike_offset);
+    
+    /**
+     * Localize threshold crossing by bisectioning.
+     * @param   double_t length of interval since previous event
+     * @returns time from previous event to threshold crossing
+     */
+    double_t bisectioning_(const double_t dt) const;
+
+    void spike_test_count_(const double_t);
+    void spike_test_(const double_t);
+    bool is_spike_(const double_t);
+
+    // ---------------------------------------------------------------- 
+
+    /** 
+     * Independent parameters of the model. 
+     */
+    struct Parameters_
+    {
+      /** Membrane time constant in ms. */
+      double_t tau_m_; 
+      
+      /** Time constant of exc. synaptic current in ms. */
+      double_t tau_ex_;
+      
+      /** Time constant of inh. synaptic current in ms. */
+      double_t tau_in_;
+      
+      /** Membrane capacitance in pF. */
+      double_t c_m_;
+      
+      /** Refractory period in ms. */
+      double_t t_ref_;
+      
+      /** Resting potential in mV. */
+      double_t E_L_;
+      
+      /** External DC current [pA] */
+      double_t I_e_;
+      
+      /** Threshold, RELATIVE TO RESTING POTENTAIL(!).
+          I.e. the real threshold is U_th_ + E_L_. */
+      double_t U_th_;
+      
+      /** Lower bound, RELATIVE TO RESTING POTENTAIL(!).
+          I.e. the real lower bound is U_min_+E_L_. */
+      double_t U_min_;
+      
+      /** Reset potential. 
+	  At threshold crossing, the membrane potential is reset to this value. 
+	  Relative to resting potential. */
+      double_t U_reset_;
+
+      double_t a1_;
```

Comment:
Variables `a1_` to `c6` that are precomputed for the `is_spike_` function are now in `calibrate`. Comments added.